src/components/ha-time-input.ts

```diff
@@ -27,7 +27,7 @@ export class HaTimeInput extends LitElement {
     const parts = this.value?.split(":") || [];
     let hours = parts[0];
     const numberHours = Number(parts[0]);
-    if (numberHours && useAMPM && numberHours > 12) {
+    if (numberHours && useAMPM && numberHours > 12 && numberHours < 23) {
```

Comment:
Since values above 12 are invalid, we should probably set it to 12 instead of not changing it.

Reply:
I don't think it's possible not to handle number between [13, 23] since this.value is always expressed in 24h format.

Reply:
I mean in am/pm mode, so above 23 in 24h mode.

Reply:
This is the render() function. I don't think we should fix invalid input in this function. If we can correctly represent the value, we should (as we do), but if it's an invalid value, we should keep it as is.
We can add more logic to the _timeChanged function, but it's beyond the original scope of this change.

Reply:
True 👍